src/custom/components/SearchModal/CurrencySearch/sorting.ts

```diff
@@ -0,0 +1,56 @@
+import { Currency, CurrencyAmount, Token } from '@uniswap/sdk-core'
+import { useMemo } from 'react'
+
+import { useAllTokenBalances } from 'state/wallet/hooks'
+
+// compare two token amounts with highest one coming first
+function balanceComparator(balanceA?: CurrencyAmount<Currency>, balanceB?: CurrencyAmount<Currency>) {
+  if (balanceA && balanceB) {
+    return balanceA.greaterThan(balanceB) ? -1 : balanceA.equalTo(balanceB) ? 0 : 1
+  } else if (balanceA && balanceA.greaterThan('0')) {
+    return -1
+  } else if (balanceB && balanceB.greaterThan('0')) {
+    return 1
+  }
+  return 0
+}
+
+function getTokenComparator(balances: {
```

Comment:
didn't this exist somewhere already? or did u just move this and mod it a bit?

Reply:
It exists, this is mod

Reply:
maybe it's worth making a const list of tokens we'd like to prioritise listed in order of importance

e.g `const PRIORITISED_TOKENS = [COW, GNO]` where lower index = higher prio